### PR TITLE
bashlib: list-resources, show-resource, and input-files

### LINKED ISSUE #751: bashlib: missing --list-resources and --show-resource

### PR DESCRIPTION
**Pull request by bertsky**
Fixes #751

### CODE REVIEW

**Comment by bertsky** [comment]
Now also fixes #571

File: ocrd/bashlib/src/wrap.bash [hunk 1]
```diff
@@ -27,4 +27,20 @@ ocrd__wrap () {
...
+    done < <(ocrd bashlib input-files \
+                  -m "${ocrd__argv[mets_file]}" \
+                  -I "${ocrd__argv[input_file_grp]}" \
+                  -O "${ocrd__argv[output_file_grp]}" \
+                  ${ocrd__argv[page_id]:+-g} ${ocrd__argv[page_id]:-})
+}
+
+# usage: pageId=$(ocrd__input_file 3 pageId)
+ocrd__input_file() {
+    eval echo "\${${ocrd__files[$1]}[$2]}"
```
Comment (kba) [inline ocrd/bashlib/src/wrap.bash:45]: ingenious use of bash :)

**Review by kba — APPROVED** [review]
Great, thank you

File: ocrd/bashlib/src/wrap.bash (hunk 1 above)
Comment (bertsky) [reply ocrd/bashlib/src/wrap.bash:45]: the usual `eval` as last resort; bash is too impoverished on the data structure side...

Comment (kba) [reply ocrd/bashlib/src/wrap.bash:45]: Yeah, it's unfortunate that there are not pointers or multi-dimensional data structures in bash...

**Pull request merged**